### ISSUE TITLE
Replication of Relation Extraction results - share finetuning parameters

### ISSUE THREAD

**Issue by ypapanik** [dmis-lab/biobert#15]
Hi,
congratulations for this really interesting work!
I am trying to replicate your results on relation extraction (so far tried GAD and euadr) and (perhaps I am missing something) I cannot replicate the numbers you report. I mostly play with the number of epochs, but what I see is:
a) largely varying metrics values across runs
b) even when I get a decent precision/recall/f1 value, if I inspect the predictions they seem quite random (all values are around 0.5).
Furthermore, I tried to run this on another popular benchmark (CDR) and again I get the same strange behaviour. 
Is it possible to share the exact parameters you used to finetune your relation extraction models?
PS: I had tried 1-2 months ago to modify BERT directly to do relation extraction, but was not very happy with the results.

**Comment by wonjininfo** [comment]
Hi,
We (authors) cross-checked the result before we made this repository public. 

We are afraid to tell you that there might be some missed points. 
Would you provide us exact command line code you executed, and the score you got so that we can reproduce the problem?

The score below is what I get from the script when I was writing README file. 
```
.tsv
recall      : 92.88%
specificity : 67.19%
f1 score    : 83.52%
precision   : 75.87%
```

Thanks,
WonJin

**Comment by ypapanik** [comment]
Hi,
thanks for replying! I can eventually replicate with small variations the numbers, my issue had to do with not deleting the model_dir before re-running the code.

**Comment by antonyscerri** [comment]
Hi

I was also trying to replicate a number of the fine tuning results. In particular for this RE task using EUADR, how many epochs did you train for? The result I've got using the same parameters as the example given for GAD appear to favour predicting the same answer for everything which is clearly wrong and won't yield results anything close to those you reported.

Also for the BioASQ task could you tell me where the data for the Task 6B is available to repeat your results?

Thanks

Tony

**Comment by wonjininfo** [comment]
Hi Tony,
I've just double checked the code and executed the code again. 
I've used same settings from README.
```
--max_seq_length=128 \
--train_batch_size=32 \
--learning_rate=2e-5 \
--num_train_epochs=3.0 \
```
The score below is what I reproduced from the code.
```
Precision : 78.66455
Recall : 93.04636
F1 Score : 84.82636
```
(Averaged 10 scores from 10 sub-datasets)

There are a few points you might missing 
* Delete or alter `output_dir` everytime you execute the script.
* Use `python ./biocodes/re_eval.py` and `test_results.tsv` for evaluation. `eval_results.txt` on `output_dir` doesn't mean anything.
* For GAD and EUADR datasets are split into 10 subdatasets. You shoud use corresponding answers for evaluation. For example, 
```
python ./biocodes/re_eval.py --output_path=euadr/9/test_results.tsv --answer_path=euadr/9/test.tsv
```
Thanks,
WonJin

**Comment by antonyscerri** [comment]
Hi

Yes I have been starting with empty directories each time, and for GAD i was able to use those settings and get some what close results to yours (in the ReadMe). Were you able to reproduce similar results for EUADR?

For EUADR though the results were kind of irrelevant as the model was predicting label 1 for everything (using both BERT base and BioBERT initialisation), and that was using the test_results.tsv, both models results were (and I did switch the task_name between gad and euadr, although from the code it looks like you can use either in this case):

f1 score    : 84.37%
recall      : 100.00%
precision   : 72.97%
specificity : 0.00%

I tried increasing the epochs to 30 for EUADR and things look a bit better, for BERT I get:
f1 score    : 82.54%
recall      : 96.30%
precision   : 72.22%
specificity : 0.00%

And BioBERT:
f1 score    : 81.97%
recall      : 92.59%
precision   : 73.53%
specificity : 10.00%

Given the EUADR test set 1 has an imbalance of labels to predict of 0 - 10 and 1 - 27 (as it appeared in most of the splits) simply predicting 1 for everything is how it achieved my first results above. I should also mention I've been testing this using the PubMed+PMC BioBERT pretrained model.

Thanks

Tony
